src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,97 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_SIZE = 10;
+    private static final double CAPACITY_INDEX = 1.5;
+    private int size;
+    private T[] elements;
+
+    public ArrayList() {
+        elements = (T[]) new Object[DEFAULT_SIZE];
+    }
+
     @Override
     public void add(T value) {
-
+        resizeArray();
+        elements[size++] = value;
```

Comment:
The 'resizeArray' method name is misleading because the array is not always resized when this method is called. Consider renaming it to 'ensureCapacity' or 'growIfNecessary' to better reflect its purpose.